plugins/auth-backend-module-microsoft-provider/src/strategy.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { PassportProfile } from '@backstage/plugin-auth-node';
+import { decodeJwt } from 'jose';
+import fetch from 'node-fetch';
+import { Strategy as MicrosoftStrategy } from 'passport-microsoft';
+
+export class ExtendedMicrosoftStrategy extends MicrosoftStrategy {
+  userProfile(
+    accessToken: string,
+    done: (err?: Error | null, profile?: PassportProfile) => void,
+  ): void {
+    if (this.skipUserProfile(accessToken)) {
+      done(null, undefined);
+      return;
+    }
+
+    super.userProfile(
+      accessToken,
+      async (err?: Error | null, profile?: PassportProfile) => {
+        if (profile) {
+          profile.photos =
+            profile.photos ?? (await this.getProfilePhotos(accessToken));
+        }
+        done(err, profile);
+      },
+    );
+  }
+
+  private skipUserProfile(accessToken: string): boolean {
+    const { aud, scp } = decodeJwt(accessToken);
+    const hasGraphReadScope =
+      aud === '00000003-0000-0000-c000-000000000000' &&
+      (scp as string)
+        .split(' ')
+        .map(s => s.toLowerCase())
+        .includes('user.read');
```

Comment:
no super-scopes to look for, that are a superset of user.read?

Reply:
not that I am aware of. Just restored this behavior, though. I will check the docs to verify it.

Reply:
As of https://learn.microsoft.com/en-us/entra/identity-platform/scopes-oidc
> In requests to the authorization server, for the Microsoft identity platform, if the resource identifier is omitted in the scope parameter, the resource is assumed to be Microsoft Graph. For example, `scope=User.Read` is equivalent to `https://graph.microsoft.com/User.Read`.

Also, there is a `User.Read.All` scope.

We create the `MicrosoftStrategy` with default scopes now (previously, it was not set):
```ts
scope: ['user.read']
```

I will adjust it to check for both scopes with their short and long form.